### PR TITLE
Added Uri and Decimal support to InvariantParser.

### PR DESCRIPTION
**Pull request by jonfortescue**

### CODE REVIEW

File: src/System.Text.Primitives/System/Text/InvariantParser_decimal.cs [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+                        else
+                        {
+                            bytesConsumed = 0;
+                            return false;
+                        }
+                    }
+                }
+                else
+                {
+                    if (!digitLast)
```
Comment (joshfree) [inline src/System.Text.Primitives/System/Text/InvariantParser_decimal.cs]: do you actually need digitLast in this method?

File: tests/System.Text.Primitives.Tests/ParserTests.cs [hunk 2]
```diff
@@ -566,5 +611,59 @@ public unsafe void ParseUtf8ByteStarToBool(string text, bool expectSuccess, int
...
+        #region uri
+        [Theory]
+        [InlineData("http://www.example.com/", true, 0, "http://www.example.com/", 23)]
+        [InlineData("https://www.example.org/important.php?true=true", true, 0, "https://www.example.org/important.php?true=true", 47)]
+        [InlineData("http://msw", true, 0, "http://msw", 10)]
+        [InlineData("ftp://example.org", true, 0, "ftp://example.org", 17)]
+        [InlineData("file:///C:\\Users", true, 0, "file:///C:\\Users", 16)]
+        [InlineData("C:\\Users", true, 0, "C:\\Users", 8)]
+        [InlineData("http://&5", false, 0, "", 0)]
+        [InlineData("(*)$#*(", false, 0, "", 0)]
```
Comment (joshfree) [inline tests/System.Text.Primitives.Tests/ParserTests.cs]: could you add more negative test cases here?

File: src/System.Text.Primitives/System/Text/InvariantParser_decimal.cs [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
```
Comment (joshfree) [inline src/System.Text.Primitives/System/Text/InvariantParser_decimal.cs]: same comment

File: tests/System.Text.Primitives.Tests/ParserTests.cs (hunk 2 above)
Comment (jonfortescue) [inline tests/System.Text.Primitives.Tests/ParserTests.cs]: indeed. Will do.

File: src/System.Text.Primitives/System/Text/InvariantParser_decimal.cs (hunk 1 above)
Comment (jonfortescue) [inline src/System.Text.Primitives/System/Text/InvariantParser_decimal.cs]: Ah, no, you're right -- that's an unnecessary carry-over from float/double. Changing now.

**Pull request merged**